spec/java_buildpack/framework/checkmarx_iast_agent_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'spec_helper'
+require 'component_helper'
+require 'java_buildpack/framework/checkmarx_iast_agent'
+
+describe JavaBuildpack::Framework::CheckmarxIastAgent do
+  include_context 'with component help'
+
+  it 'does not detect without checkmarx-iast service' do
+    expect(component.detect).to be_nil
+  end
+
+  context do
+
+    before do
+      allow(services).to receive(:one_service?).with(/^checkmarx-iast$/, 'server').and_return(true)
+      allow(services).to receive(:find_service).and_return('credentials' => { 'server' => 'http://iast-server:8080/' })
+    end
+
+    it 'detects with checkmarx-iast service' do
+      expect(component.detect).to eq('http://iast-server:8080')
+    end
+
+    it 'updates JAVA_OPTS' do
```

Comment:
There needs to be a test here to ensure that compilation happens appropriately.  This would include stubbing in a fake file for download.